server/src/main/java/io/crate/metadata/cluster/OpenTableClusterStateTaskExecutor.java

```diff
@@ -113,7 +114,7 @@ protected ClusterState execute(ClusterState currentState, OpenTableRequest reque
 
             // The index might be closed because we couldn't import it due to old incompatible version
             // We need to check that this index can be upgraded to the current version
-            updatedIndexMetadata = metadataIndexUpgradeService.upgradeIndexMetadata(updatedIndexMetadata, templateMetadata, minIndexCompatibilityVersion);
+            updatedIndexMetadata = metadataIndexUpgradeService.upgradeIndexMetadata(updatedIndexMetadata, templateMetadata, minIndexCompatibilityVersion, currentState.metadata().custom(UserDefinedFunctionsMetadata.TYPE));
```

Comment:
If this is using the current state, why is the UDF update necessary? It should already be at the current state.

Reply:
Thanks, removing the ones using current state.